src/common/src/array/stream_chunk.rs

```diff
@@ -75,6 +75,50 @@ pub struct StreamChunk {
     // TODO: Optimize using bitmap
     ops: Arc<[Op]>,
     data: DataChunk,
+    pub(super) meta: Option<StreamChunkMeta>,
```

Comment:
This is a little bit diverging from the current design. Additional information such as offset, transaction id, etc., should be placed in Barrier instead of StreamChunk. 

By the way, @tabVersion and @shanicky is working on improving S3 Source, where they may need to pass a 'split completition' message throw stream. AFAIK they prefer to record this information in Barrier. Although the use cases are not exactly same - in their cases they need to report the info to Meta node, while here we need to pass information between streaming operators.

Reply:
Sorry, I misunderstood. According to our offline discussion, I realized that `meta` is actually per row. Thus, I would suggest to make it a column similar to `_row_id`. Moreover, since this meta column is only used to pass offset info from `ChangelogExecutor` to `BackfillExecutor`,  we can add this column in the output of `ChangelogExecutor` and remove it at `BackfillExecutor`, so that other operators won't be aware of it.

Reply:
Will do.

Reply:
> Thus, I would suggest to make it a column similar to _row_id.

+1 for this

Reply:
+1, just like the processing path of the `_rw_kafka_timestamp`

Reply:
fixed